### DISCUSSION TITLE
New layout system just dropped

### DISCUSSION THREAD

**Discussion by Ottatop** [pinnacle-comp/pinnacle#178] [Dev log]
What is poppin' my Sky Shredders and Flying Fortresses,

We've got a new layout system! It's built on the same idea that powers [River](https://github.com/riverwm/river)'s layout generators, namely that you now have the ability to tell the compositor the specific geometries of windows for layouting.

As a result, the old layout system has been removed and old configs will break.

As a plus, we've now got window gaps!
![image](https://github.com/pinnacle-comp/pinnacle/assets/120758733/974d0605-1420-4ea1-8542-1780eaa23dab)

The system has been designed to be user configurable, so you can write up your own layouts. Unfortunately there are no docs on how to do that yet sooo I guess go dig around in the source code for now?

Yes, it's a little rough around the edges i.e. I don't think you can change layout settings dynamically in the Rust API yet, but hey it's a whole lot better than what we had before! And it now also has Awesome's fair layout, which is weird because that's the only layout I use and I've *just* now only added it.

Also sloppy focus was added who knows how long ago

Now if you'll excuse me, I've got exams that I've been procrastinating studying on :upside_down_face: 

Edit: oh yea the Lua API reference is borked lmao

**Comment by ildar** [comment]
This is great that Pinnacle's progressing. Yet, more important, good luck
and enjoy your studies.
